### ISSUE TITLE
SNOWCLI issues running with python 3.8

### ISSUE THREAD

**Issue by orellabac** [Snowflake-Labs/snowcli#39]
Steps:

python --version
Python 3.8.12

$uname -a
Linux 4ae3befb-2ddd46-2d4855-2dab9a-2d4748d90ba440-621 5.4.0-1091-azure #96~18.04.1-Ubuntu SMP Tue Aug 30 19:15:32 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

pip install snowflake-cli-labs

but when I run
snow -h

I get:
```
Traceback (most recent call last):
  File "/home/BlackDiamond/.local/bin/snow", line 5, in <module>
    from snowcli.cli import app
  File "/home/BlackDiamond/.local/lib/python3.8/site-packages/snowcli/cli/__init__.py", line 5, in <module>
    from snowcli.config import AppConfig
  File "/home/BlackDiamond/.local/lib/python3.8/site-packages/snowcli/config.py", line 6, in <module>
    from snowcli.snow_connector import SnowflakeConnector
  File "/home/BlackDiamond/.local/lib/python3.8/site-packages/snowcli/snow_connector.py", line 10, in <module>
    class SnowflakeConnector():
  File "/home/BlackDiamond/.local/lib/python3.8/site-packages/snowcli/snow_connector.py", line 34, in SnowflakeConnector
    def createFunction(self, name: str, inputParameters: str, returnType: str, handler: str, imports: str, database: str, schema: str, role: str, warehouse: str, overwrite: bool, packages: list[str])  -> SnowflakeCursor:
TypeError: 'type' object is not subscriptable
```

This seem to be because python 3.8 does not support typing decorations for generics those were introduced in Python 3.9

According to https://docs.snowflake.com/en/developer-guide/snowpark/python/setup.html

Minimum requirements are Python 3.8

**Comment by sfc-gh-jhollan** [comment]
Thanks @orellabac for trying and reporting. Yes unfortunately it is using typing right now as you mentioned, so if using the pip approach requires Python 3.9 or greater. I think some `switch` statements in there too which are 3.9/3.10+ required. Depending on pain we could look at moving down to 3.8 (esp. as Snowpark right now supports 3.8).  

In the meantime, we do generate Linux binaries with Python 3.10 during the automated build system, but not yet deployed. I will try to do some experimentation on if we could allow that as an install method for Linux that would get around this (you can get around on mac by using `homebrew`, not sure if homebrew + Linux could mitigate as well

**Comment by sfc-gh-jhollan** [comment]
Ok @orellabac if interested in running this on Linux as a "single binary" executable (so shouldn't matter even if you don't have a version of Python on your machine), just tested this out on Linux and it worked:

```
apt update && apt install wget zip -y
mkdir snowcli
cd snowcli
# install libssl - may not be required but was on my docker image
wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.16_amd64.deb
sudo dpkg -i libssl1.1_1.1.1f-1ubuntu2.16_amd64.deb
# install the latest linux release - this is 0.1.7 at time of posting
wget https://github.com/Snowflake-Labs/snowcli/releases/download/v0.1.7/snow-linux.zip
unzip snow-linux.zip
# link to the executable
ln -s $(pwd)/snow-linux/snow /usr/local/bin
```
And you should be able to now say `snow --version` and have it work

**Comment by orellabac** [comment]
Thanks that is a good alternative. I will take a look at the code to see how hard is to have it working on 3.8 / 3.9. / 3.10 and 3.11. Is there any functionality that relies on typing?

**Comment by sfc-gh-jhollan** [comment]
Nope - was purely to help with IDE typing / validation. Happy to remove if it makes more backward compatible

**Comment by sfc-gh-asaini** [comment]
@sfc-gh-jhollan : Went through another user research test today and felt the need for 3.8 or 3.9 support

**Comment by sfc-gh-jhollan** [comment]
resolved in 0.1.8 just released

**Comment by tomas-sobotik** [comment]
I was facing same issue in Python 3.8.16, but it works in 3.8.15. Tested with the latest snowcli version 0.1.21. Probably something has changed in 3.8.16 and bring this issue back.

**Comment by jeffhollan** [comment]
Do you know what you were running in 0.1.21 when error showed up? Wonder if related to this command / change we are patching https://github.com/Snowflake-Labs/snowcli/pull/82

**Comment by blackary** [comment]
This should be fixed in the next release, by #82

**Comment by sfc-gh-jhollan** [comment]
Thanks all - 0.1.22 is out now and should be fixed
